pkg/workflows/wasm/host/module.go

```diff
@@ -580,7 +587,49 @@ func runWasm[I idMessage, O proto.Message](
 }
 
 func (m *module) SetCapabilityExecutor(handler CapabilityExecutor) error {
-	return errors.New("not implemented")
+	callCapAsync := m.cfg.CallCapability
+	if handler != nil {
+		callCapAsync = toCallCapAsync(m.capabilityCallStore, handler)
+	}
+
+	// shadow any existing implementation with the one passed by the call
+	if err := m.linker.FuncWrap(
+		"env",
+		"call_capability",
+		createCallCapFn(m.cfg.Logger, m.executeStore, callCapAsync),
+	); err != nil {
+		return fmt.Errorf("error wrapping callcap func: %w", err)
+	}
```

Comment:
As an FYI, it shouldn't change executables that are already running.

Why not have the linker take the module instead and set the handler on the module. Then you can link aginst the module and the callback will always use the latest executor.